doc/sphinx/addendum/extraction.rst

```diff
@@ -413,6 +420,98 @@ As an example of translation to a non-inductive datatype, let's turn
+OCaml code with C code, the linker needs to know
+
+   * which C functions will be called by the ML code (external)
+   * which ML functions shall be accessible by the C code (callbacks)
+
+.. cmd:: Extract Foreign Constant @qualid => @string
+
+   Like :cmd:`Extract Constant`, except that the referenced ML terms
+   will be declared in the form
+
+   ``external @qualid : ML type = "@string"``.
+
+.. caution::
+
+   * The external function name :n:`@string` is not checked in any way.
+
+   * The user must ensure that the C functions given to realize the axioms have
+     the expected or compatible types. In fact, the strings containing realizing
+     code are just copied to the extracted files.
+
+.. exn:: Extract Foreign Constant is supported only for OCaml extraction.
+
+   Foreign function calls are only supported for OCaml.
+
+.. exn:: Extract Foreign Constant is supported only for functions.
+
+   This error is thrown if @qualid is of sort ``Type`` and external functions only
+   work for functions.
+
+.. exn:: The term @qualid is already defined as inline custom constant.
+
+   The :n:`@qualid` was previously used in a
+   :cmd:`Extract Inlined Constant` command. Using :cmd:`Extract Foreign Constant`
+   for :n:`@qualid` would override this command.
+
+.. cmd:: Extract Callback {? @string } @qualid
+
+   This command makes sure that after extracting the :term:`constants <constant>`
+   specified by :n:`@qualid`, a constant ML function will be generated that
+   registers :n:`@qualid` as callback, callable by :n:`@string`.
+   This is done by declaring a function
+   ``let _ = Callback.register "@string" @qualid``
+   This expression signals OCaml that the given ML function :n:`@qualid` shall be
+   accessible via the alias :n:`@string`, when calling from C/C++.
+   If no alias is specified, it is set to the string representation of :n:`@qualid`.
+
+.. caution::
+   * The optional alias :n:`@string` is currently **not** checked in any way.
+
+   * The user must ensure that the callback aliases are
+     unique, i.e. when multiple modules expose a callback, the user should make
+     sure that no two :n:`@qualid` share the same alias.
+
+.. note::
+   Using Extract Callback has no impact on the rest of the synthesised code since
+   it is an additional declaration. Thus, there is no impact on the correctness
+   and type safety of the generated code.
+
+.. exn:: Extract Callback is supported only for OCaml extraction.
+
+   The callback registration mechanism ``Callback.register`` is specific
+   to OCaml. Thus, the command is only usable when extracting OCaml code.
+
+.. cmd:: Print Extraction Foreign
+
+   Prints the current set of custom foreign functions
+   declared by the command :cmd:`Extract Foreign Constant` together with its
+   associated foreign ML function name.
+
+.. .. cmd:: Reset Extraction Foreign
+
+..   Resets the set of custom externals
+..   declared by the command :cmd:`Extract Foreign Constant`.
```

Comment:
These are comments, not visible in the doc.  Is that intended and needed?

Reply:
Yes, @jfehrle, his is intended.
See my [comment](https://github.com/coq/coq/pull/18270/#issuecomment-1810268536). It is not clear yet, how to handle the resetting of (inlined/foreign) customs, i.e. removing a (global) substitution for a qualid. This is not a problem of this PR but rather a general one which should be addressed separately. Thus I deactivated the functionality and documentation for `Reset Extraction Foreign` for now.